### DISCUSSION TITLE
Save credentials using git config --global user.name

### DISCUSSION THREAD

**Discussion by TheSupremeGod** [community/community#22327] [New to GitHub]
<p>I don’t want git to ask me for my username (or even my password) every time i push, pull or clone.</p>
<p>Whenever I clone a personal repository i already do <a href="https://gihub.com/Username/Repo" rel="nofollow noopener">https://gihub.com/Username/Repo</a> so it won’t ask me there but if I do</p>
<pre><code>git config --global user.name "Username"
</code></pre>
<p>and try to push to that same repository it promts me to enter in anyway.</p>
<p>How do I prevent that?</p>

**Comment by Wabri** [comment, marked as answer]
<p>You can use:</p>
<pre><code>git config --global credential.helper cache
</code></pre>
<p> This command prevent git to ask username and password, not forever but with a default limit to 15 minutes.</p>
<p>To change this limit you can run:</p>
<pre><code>git config --global credential.helper 'cache --timeout=3600'
</code></pre>
<p>This move the default limit to 1 hours.</p>
<p>You can find all this information on <a href="https://help.github.com/articles/caching-your-github-password-in-git/" rel="nofollow noopener">help.github</a>.</p>
<p>Hope this helps :slightly_smiling_face:  </p>
<p>-Gabriele-</p>
<blockquote>
<p><strong>Mark helpful posts with Accept as Solution to help other users locate important info. Don’t forget to give Kudos for great content!</strong></p>
</blockquote>

**Comment by TheSupremeGod** [comment]
<p>So the user.name has no effect on pushing or pulling? That’s unfortunate but okay.</p>

**Comment by Wabri** [comment]
<p>The link that I gave you said:<br>
“…credential helper to tell Git to remember your GitHub username and password every time it talks to GitHub.”<br>
So I think even username is cached.</p>
